### ISSUE TITLE
Option to make Obsidian the entire sidebar without requiring a root node.

### ISSUE THREAD

**Issue by ChildishhAlbino** [HiDeoo/starlight-obsidian#5]
### Is your feature request related to a problem?

I want to use Obsidian as the entirety of my Astro site, I have been hacking together my own sub-par version of this but want to switch to this ASAP. It is small annoyance that I can't seem to have my obsidian subfolders be the root nodes of the sidebar and must have them nested under a singular root node.

### Describe the solution you'd like

A configuration option `disableRootNode: true | false` to tell the plugin to not generate the subfolders under a root node.

### Describe alternatives you've considered

I've tried looking at the `obsidianSideBarGroup` array but it's empty at the time I can access it before startlight/astro have started up.

### Additional Context

Happy to help code this with a pointer or two if you're too busy on more important things.

**Comment by HiDeoo** [comment]
Thanks for the feedback, I think this makes sense. Unfortunately, I don't think this is something I can implement at the moment as this would require a change in Starlight.

The current sidebar generation relies on Starlight [autogenerated groups](https://starlight.astro.build/guides/sidebar/#autogenerated-groups) which are always a group.
The only possible workaround at the moment would be to duplicate the sidebar generation code from Starlight and manually generating it from the plugin which is not ideal.

I think this is a valid use case that I have already encountered multiple times so I think the best solution would be to open a feature request on the Starlight repository and see if this is something that can be implemented in the future.
I'll keep thinking a bit more about this, list various use cases that could benefit from this and how API-wise this could be supported in Starlight. When I have a clearer idea, I will propose a solution there, and implement it if this is something of interest.

**Comment by ChildishhAlbino** [comment]
Hmm - what if the user provided a folder depth underneath the root node to the plugin and then it could run the file tree, work out which folders should be at the "root" and then return back multiple groups for each folder at that depth.

Happy to draw a diagram if that isn't clear enough?

**Comment by nmboris** [comment]
Couldn't you achieve this through simple configuration?

In my first tests, this worked perfectly with both referenced images and links.

I simply referenced the subdirectories from the Obsidian root directly in `astro.config.mjs` and commented out the `obsidianSidebarGroup`...

<img width="739" alt="SCR-20240424-nmgq" src="https://github.com/HiDeoo/starlight-obsidian/assets/63914278/b269b3d4-14fb-4c68-8247-2654d15fc321">
<img width="374" alt="SCR-20240424-nnfm" src="https://github.com/HiDeoo/starlight-obsidian/assets/63914278/8df61259-f81a-4016-a421-ef9664717c38">
<img width="373" alt="SCR-20240424-nmwr" src="https://github.com/HiDeoo/starlight-obsidian/assets/63914278/c8468411-5972-467a-abf3-61142bca3b2e">

**Comment by ChildishhAlbino** [comment]
@nmboris that's a creative way to get around it indeed - let me try that and see the results in my repo :D

**Comment by HiDeoo** [comment]
> Couldn't this be achieved simply through configuration?

This is indeed doable through configuration, altho it has several downsides depending on who is in charge of handling the configuration:

- If it's the user, that means they need to learn about the configuration options and how to use them and manually set them up. It also does not scale well depending on the number of root folders and files.
- If it's the plugin, this adds more complexity for plugin authors.

Hitting this limitation with multiple plugins is what is making me think that a Starlight solution would be beneficial for everyone.

**Comment by ChildishhAlbino** [comment]
> If it's the user, that means they need to learn about the configuration options and how to use them and manually set them up. It also does not scale well depending on the number of root folders and files.

Yeah have to agree with this - even trying to maintain this for a few folders was annoying enough so I can't image if I had to maintain more than 5 if it would be worth it - I decided to write my own function that spat out autogenerated group of the subfolders of the root of my vault - works a treat and lets me decorate the titles with emojis which is nice.

Happy to close this issue if you think this is better solved within starlight @HiDeoo

**Comment by HiDeoo** [comment]
> Happy to close this issue if you think this is better solved within starlight @HiDeoo

I would prefer to keep it open. 

If Starlight implement something to help with this issue, it will be a good reminder to update the plugin and add support for this. If not, it will also be a good reminder that the plugin should be updated to solve this issue as I still think it's a totally valid use-case encountered already a few times.

**Comment by sanabel-al-firdaws** [comment]
hey this works ig then you need to remove the sidebar

```ts

  starlightObsidian({
          output: "",
          vault: "./vault",
        }),


....

// sidebar: [obsidianSidebarGroup],
```

**Comment by anotheri** [comment]
@sanabel-al-firdaws unfortunately no, such config removes all the content from `public` folder
